pkg/products/amqonline/reconciler.go

```diff
@@ -412,6 +419,51 @@ func (r *Reconciler) reconcileAddressSpacePlans(ctx context.Context, serverClien
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
+func (r *Reconciler) reconcileServiceAdmin(ctx context.Context, serverClient k8sclient.Client, serviceAdminRole *rbacv1.Role) (integreatlyv1alpha1.StatusPhase, error) {
+	r.logger.Info("reconciling service admin role to the dedicated admins group")
+
+	_, err := controllerutil.CreateOrUpdate(ctx, serverClient, serviceAdminRole, func() error {
+		owner.AddIntegreatlyOwnerAnnotations(serviceAdminRole, r.inst)
+
+		serviceAdminRole.Rules = GetServiceAdminRole("").Rules
```

Comment:
```suggestion
		serviceAdminRole.Rules = GetServiceAdminRole(r.Config.GetNamespace()).Rules
```

^I presume it needs a valid namespace, not sure.

Also might be easier to just say the following at this point, and get rid of the `GetServiceAdminRole` function?

```suggestion
		serviceAdminRole.Rules = []rbacv1.PolicyRule{
			{
				APIGroups: []string{"admin.enmasse.io"},
				Resources: []string{"addressplans", "addressspaceplans", "brokeredinfraconfigs", "standardinfraconfigs", "authenticationservices"},
				Verbs:     []string{"create", "get", "update", "delete", "list", "watch", "patch"},
			},
		}
```

Reply:
I think so too, I will refactor now.

Reply:
refactored now with tests. Amended the last commit and pushed the branch